invoices/interface.go

```diff
@@ -163,3 +164,37 @@ type InvoiceSlice struct {
 // CircuitKey is a tuple of channel ID and HTLC ID, used to uniquely identify
 // HTLCs in a circuit.
 type CircuitKey = models.CircuitKey
+
+// InvoiceUpdater is an interface to abstract away the details of updating an
+// invoice in the database. The methods of this interface are called during the
+// in-memory update of an invoice when the database needs to be updated or the
+// updated state needs to be marked as needing to be written to the database.
+type InvoiceUpdater interface {
+	// AddHtlc adds a new htlc to the invoice.
```

Comment:
Already mentioned this in the comments below, but I think these functions need a more detailed name because by just looking at the current naming I would not be able to explain why the kv-value implementation doesn't need an AddHtlc function, because htlcs are added as well ?